src/reduce_dims.cpp

```diff
@@ -0,0 +1,117 @@
+#include <migraphx/reduce_dims.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+bool reduce_dim(std::vector<shape>& shapes, std::size_t n)
+{
+    std::vector<std::size_t> new_lens;
+    for(const auto& s : shapes)
+    {
+        if(n >= s.lens().size())
+            return false;
+        if((n + 1) >= s.lens().size())
```

Comment:
is this if statement necessary? if `n >= s.lens().size()`, then that implies `n+1` is also `>= s.lens().size()`

Reply:
The if statement isnt checked when `n >= s.lens().size()` is true but when its not true. It could be that the first `if` statement is unnecessary(in fact there is no code coverage for it).

Reply:
Yeah I think I had it backwards as well. It should be that the first one is not necessary.

Reply:
I removed the `if` statement.